projects/vdk-plugins/vdk-kerberos-auth/src/vdk/plugin/kerberos/kerberos_plugin.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+import logging
+from typing import List
+from typing import Union
+
+from vdk.api.plugin.hook_markers import hookimpl
+from vdk.api.plugin.plugin_registry import IPluginRegistry
+from vdk.internal.builtin_plugins.run.job_context import JobContext
+from vdk.internal.control.exception.vdk_exception import VDKException
+from vdk.internal.core.config import ConfigurationBuilder
+from vdk.plugin.kerberos.authenticator_factory import KerberosAuthenticatorFactory
+from vdk.plugin.kerberos.base_authenticator import BaseAuthenticator
+from vdk.plugin.kerberos.kerberos_configuration import add_definitions
+from vdk.plugin.kerberos.kerberos_configuration import KerberosPluginConfiguration
+
+log = logging.getLogger(__name__)
+
+
+class KerberosPlugin:
+    def __init__(self):
+        self.__authenticator_factory = None
+        self.__authenticator = None
+
+    @staticmethod
+    @hookimpl
+    def vdk_configure(config_builder: ConfigurationBuilder) -> None:
+        add_definitions(config_builder)
+
+    @hookimpl(tryfirst=True)
+    def initialize_job(self, context: JobContext) -> None:
+        kerberos_configuration = KerberosPluginConfiguration(
+            context.name, str(context.job_directory), context.core_context.configuration
+        )
+        self.__authenticator_factory = KerberosAuthenticatorFactory(
+            kerberos_configuration
+        )
+        authenticator = self.__get_authenticator(
+            kerberos_configuration.authentication_type()
```

Comment:
this seems too complex - keeping the factory instance as a class field, then on the next line using it to produce an authenticator - that is externalised in a dedicated method by itself...
This could be done locally within the scope of initialize_job(), simplifying the references and abstractions, to make it more readable.

Reply:
Agreed. I've refactored this code for simplicity.